### PR TITLE
Fixes related to recent preferences and workspaces changes

### PR DESCRIPTION
**Pull request by dmitrio95**
This pull request contains fixes for two issues appeared after merging #4079. The first commit removes redundant operations on updating styles and icons and moves `updateIcons()` call to the more appropriate place. One of these operations, namely updating the stylesheet [here](https://github.com/musescore/MuseScore/commit/9fb454967697737f1d86ad30f2736d1bc0532083#diff-e1c31b5b3c39484d8087f34ad3afdcf6R1124) leads to a crash described [in the issue tracker](https://musescore.org/en/node/277697). Unfortunately I am not currently able to describe in details why it happens but since that call should anyway be redundant (as it is done later inside `MuseScore::updateUiStyleAndTheme`) I believe we can remove it from that place.

The second commit fixes a problem with workspaces translations which also [led to MuseScore crash](https://musescore.org/en/node/277747).

### CODE REVIEW

File: mscore/jackaudio.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include "seq.h"
 #include "libmscore/score.h"
 #include "libmscore/repeatlist.h"
+#include "libmscore/sig.h"
```
Comment (Jojo-Schmitz) [inline mscore/jackaudio.cpp]: Why is this needed?

Comment (dmitrio95) [reply mscore/jackaudio.cpp]: Well, perhaps all these include changes are really better not to mix with this PR. I'll remove it now from here.

Comment (Jojo-Schmitz) [reply mscore/jackaudio.cpp]: the removals looks OK to me though
